File: BST/js/traversals.js
const print_in_order = tree => {
  if (tree == null) return;
  print_in_order(tree.left);
  process.stdout.write(`${tree.value} `);
  print_in_order(tree.right);
};

const print_pre_order = tree => {
  if (tree == null) return;
  process.stdout.write(`${tree.value} `);
  print_pre_order(tree.left);
  print_pre_order(tree.right);
};

const print_post_order = tree => {
  if (tree == null) return;
  print_post_order(tree.left);
  print_post_order(tree.right);
  process.stdout.write(`${tree.value} `);
};

module.exports = { print_in_order, print_post_order, print_pre_order };

File: AVL/c/normal/traversals.c
#include <stdio.h>
#include "traversals.h"

void print_in_order(Node_ptr tree)
{
  if (tree == NULL)
    return;
  print_in_order(tree->left);
  printf("%d ", tree->value);
  print_in_order(tree->right);
}

void print_pre_order(Node_ptr tree)
{
  if (tree == NULL)
    return;
  printf("%d ", tree->value);
  print_pre_order(tree->left);
  print_pre_order(tree->right);
}

void print_post_order(Node_ptr tree)
{
  if (tree == NULL)
    return;
  print_post_order(tree->left);
  print_post_order(tree->right);
  printf("%d ", tree->value);
}